docs/data/data-grid/rows/rows.md

```diff
@@ -17,18 +17,34 @@ Otherwise, the grid will re-apply heavy work like sorting and filtering.
 
 {{"demo": "RowsGrid.js", "bg": "inline"}}
 
-:::warning
-Each row object should have a field that uniquely identifies the row.
-By default, the grid will use the `id` property of the row. Note that [column definition](/x/react-data-grid/column-definition/) for `id` field is not required.
+## Row identifier
+
+Each row must have a unique identifier.
+
+This identifier is used internally to identify the row in the various models—for instance, the row selection model—and to track the row across updates.
 
-When using dataset without a unique `id` property, you can use the `getRowId` prop to specify a custom id for each row.
+By default, the data grid looks for a property named `id` in the data set to get that identifier.
+
+If the row's identifier is not called `id`, then you need to use the `getRowId` prop to tell the grid where it's located.
+
+The following demo shows how to use `getRowId` to grab the unique identifier from a property named `internalId`:
```

Comment:
Are we sure about the number of line breaks? It looks dense and hard to skim read:

<img width="774" alt="Screenshot 2022-06-14 at 13 07 20" src="https://user-images.githubusercontent.com/3165635/173563677-3b1aa5ab-fd94-464d-bf03-42ee6ffd9a20.png">

Reply:
We can improve that one in https://github.com/mui/mui-x/pull/5195 
I felt like it was already an improvement and could be shiped as is if needed

Reply:
I would personally enjoy reading this:

<img width="779" alt="Screenshot 2022-06-14 at 13 09 23" src="https://user-images.githubusercontent.com/3165635/173563974-6b4a9ce3-2a4c-44f8-bb5a-b7baee5a0abb.png">

Reply:
> I felt like it was already an improvement and could be shiped as is if needed

@flaviendelangle For sure 👍

Reply:
Both of these look good to me, so I'm happy to accommodate either style. I would say that in general, I don't think paragraphs should be any longer than they are in the latter version. With dense material like this, it's easy to get lost in the middle of a paragraph.

I think this paragraph is the ideal length for text like this:

<img width="683" alt="Screen Shot 2022-06-14 at 9 11 03 AM" src="https://user-images.githubusercontent.com/71297412/173598659-eb68fd74-d397-4644-ad77-96056856a78a.png">

So we could say that as a general rule, you should consider a line break after 2 sentences or ~200 characters.

Reply:
cc @mbrookes for perspective on the matter.

---

If we were to look for rules that could work in practice, I would personally imagine that above 500 characters, a paragraph is either too verbose and should be broken down, or it's actually covering a different matter and hence should be broken down too. In the docs of Stripe (one I would assume receives a lot of care), it's common to find https://stripe.com/docs/products-prices/how-products-and-prices-work paragraphs of almost 400 characters but It's hard to find find one above 500.

Reply:
I'm with @oliviertassinari on this one. The second example is superior to the first. But if they needed to be to convey the concept, each paragraph could comfortably be twice as long. I don't think that there should be an arbitrary rule on paragraph length – paragraphs aren't measured in words or sentences; but I agree that verbosity should be eliminated, and that paragraphs should cover one narrow topic.